lib/rspec/core/world.rb

```diff
@@ -21,6 +21,16 @@ def initialize(configuration=RSpec.configuration)
         configuration.world = self
         @example_groups = []
         @example_group_counts_by_spec_file = Hash.new(0)
+        filter_examples
+      end
+
+      # @api private
```

Comment:
If you depend on this, we should make this `@api public` so it is subject to SemVer.

Reply:
Okay, I saw that most things were marked private and followed suit.

Reply:
`@api private` means that while a method is publicly visible according to ruby's visibility rules (e.g. so that code outside the class can call it), we do _not_ consider it part of our public API, and it can change without warning in any RSpec release with no warning whatsoever.  Which means that if you write an extension that depends upon an `@api private` method your extension may break at any time in a future RSpec release.

`@api public` means we consider it part of our public API and commit to only changing it in a major version bump, if ever.  Given you are writing code that depends on this, I presume you would like that guarantee?

Reply:
To expand on that a bit futher: most things are private here because we want to minimize the surface of the API we are committing to supporting.  The smaller the API, the easier it is to support :).

Reply:
Yes a public API would be great. Appreciate you caring.